src/eduid_webapp/reset_password/schemas.py

```diff
@@ -68,18 +68,13 @@ class ResetPasswordExtraSecSchema(CSRFRequestMixin):
     phone_index = fields.Integer(required=True)
 
 
-class ResetPasswordWithCodeSchema(CSRFRequestMixin):
-    
+class ResetPasswordExtraSecTokenSchema(CSRFRequestMixin):
+
     code = fields.String(required=True)
-    password = fields.String(required=True)
+    token_data = fields.String(required=True)
 
-    @validates('password')
-    def validate_password(self, value):
-        # Set a new error message
-        try:
-            self._validate_password(value)
-        except ValidationError:
-            raise ValidationError(_('Please use a stronger password'))
+
+class PasswordMixin(object):
```

Comment:
Why do you create a new mixin instead of using the PasswordSchema from eduid-common?

Reply:
Also it seems like the call to zxcvbn is missing the user_inputs argument.